components/post_view/post_list_row/index.ts

```diff
@@ -11,11 +11,42 @@ import {emitShortcutReactToLastPostFrom} from 'actions/post_actions';
 
 import {GlobalState} from 'types/store';
 
-import PostListRow from './post_list_row';
+import {getUsage} from 'mattermost-redux/selectors/entities/usage';
+import {getCloudLimits, getCloudLimitsLoaded} from 'mattermost-redux/selectors/entities/cloud';
+import {getLimitedViews} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/common';
 
-function mapStateToProps(state: GlobalState) {
+import {PostListRowListIds} from 'utils/constants';
+
+import PostListRow, {PostListRowProps} from './post_list_row';
+
+type OwnProps = Pick<PostListRowProps, 'listId'>
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     const shortcutReactToLastPostEmittedFrom = getShortcutReactToLastPostEmittedFrom(state);
-    return {shortcutReactToLastPostEmittedFrom};
+    const usage = getUsage(state);
+    const limits = getCloudLimits(state);
+    const limitsLoaded = getCloudLimitsLoaded(state);
+
+    const props: Pick<
+    PostListRowProps,
+    'shortcutReactToLastPostEmittedFrom' | 'usage' | 'limits' | 'limitsLoaded' | 'exceededLimitChannelId' | 'firstInaccessiblePostTime'
+    > = {
+        shortcutReactToLastPostEmittedFrom,
+        usage,
+        limits,
+        limitsLoaded,
+    };
+    if ((ownProps.listId === PostListRowListIds.OLDER_MESSAGES_LOADER || ownProps.listId === PostListRowListIds.CHANNEL_INTRO_MESSAGE) && limitsLoaded) {
+        const currentChannelId = getCurrentChannelId(state);
+        const firstInaccessiblePostTime = getLimitedViews(state).channels[currentChannelId];
```

Comment:
Totally possible I'm missing a key piece, but if this is scoped to a specific channelId, wouldn't that mean the messages limit is being applied _per channel_, rather than for the entire server?

Reply:
Good question! [Prior work](https://github.com/mattermost/mattermost-server/pull/20362) added a recurring job to calculate the latest accessible post time, as well as filtering to posts retrieval. So once the posts are fetched from the store, they are compared against that server wide last accessible post time, and any posts that are beyond the limit are filtered. Different channels will have different dates for when posts in *their* channel are filtered, but its still a single server-wide limit being applied to any channel's posts.

Reply:
Yep! That is correct.

Reply:
Ok thanks for the clarification